angular-http-server.js

```diff
@@ -1,14 +1,45 @@
 #!/usr/bin/env node
 
-var http = require("http");
 var fs = require("fs");
 var argv = require('minimist')(process.argv.slice(2));
 
-var server = http.createServer(function (req, res) {
+var server;
+if (argv.ssl || argv.https) {
+    var pem = require('pem');
+    var https = require('https');
+    pem.createCertificate({days: 1, selfSigned: true}, function(err, keys) {
+        var options = {
+            key: keys.serviceKey,
+            cert: keys.certificate,
+            rejectUnauthorized: false
+        };
+        server = https.createServer(options, requestListener);
+        start();
+    });
+}
+else {
+    var http = require("http");
+    server = http.createServer(requestListener);
+    start();
+}
+
+function requestListener(req, res) {
     // console.log(req.url);
     var url = req.url.split('?')[0]
     var possibleFilename = url.slice(1) || "dummy";
 
+    if (argv.cors) {
+      res.setHeader('Access-Control-Allow-Origin', '*');
+      res.setHeader('Access-Control-Request-Method', '*');
+      res.setHeader('Access-Control-Allow-Methods', 'OPTIONS, GET');
+      res.setHeader('Access-Control-Allow-Headers', 'authorization, content-type');
+      if ( req.method === 'OPTIONS' ) {
+        res.writeHead(200);
+        res.end();
```

Comment:
More on the OPTIONS preflight in CORS: https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/OPTIONS#Preflighted_requests_in_CORS